packages/sonarwhal/src/lib/utils/resource-loader.ts

```diff
@@ -12,62 +12,59 @@
  */
 
 import * as path from 'path';
-import { promisify } from 'util';
 
 import * as globby from 'globby';
-import * as npm from 'npm';
-import * as esearch from 'npm/lib/search/esearch';
-import * as inquirer from 'inquirer';
+import * as semver from 'semver';
 
-import { findNodeModulesRoot, findPackageRoot, readFile } from './misc';
-import { debug as d } from './debug';
-import { IConnectorBuilder, IFormatter, NpmPackage, Parser, Resource, IRuleBuilder } from '../types';
-import { validate as validateRule } from '../config/config-rules';
-import { installPackages } from './npm';
+import { getPackage, getSonarwhalPackage, findNodeModulesRoot, findPackageRoot, readFile } from '../utils/misc';
+import { debug as d } from '../utils/debug';
+import { Resource, IRuleConstructor, SonarwhalResources } from '../types';
+import { SonarwhalConfig } from '../config';
 
 const debug: debug.IDebugger = d(__filename);
 const SONARWHAL_ROOT: string = findPackageRoot();
 const NODE_MODULES_ROOT: string = findNodeModulesRoot();
-const npmLoadAsync = promisify(npm.load);
 
 /** Cache of resource builders, indexex by resource Id. */
 const resources: Map<string, Resource> = new Map<string, Resource>();
 
 /** Cache of resources ids. */
 const resourceIds: Map<string, Array<string>> = new Map<string, Array<string>>();
 
+/**
+ * Validates if a given package can be used with the current `sonarwhal` version
+ * by looking at its `peerDependencies`
+ */
+const isVersionValid = (resourcePath: string): boolean => {
+    try {
+        const pkg = getPackage(resourcePath);
+        const sonarwhalPkg = getSonarwhalPackage();
+
+        return semver.satisfies(sonarwhalPkg.version, pkg.peerDependencies.sonarwhal);
+    } catch (e) {
+        // We failed to load the package.json so it's a core resource
+        debug(e);
+
+        return true;
+    }
+};
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
 /** The type of resource */
 export const TYPE = {
```

Comment:
We should move TYPE to the folder `enums` and change it for an `enum`, and change the name